### DISCUSSION TITLE
Is there any difference between subscribing with wildcards and subscribing individually when subscribing to a large number of topics?

### DISCUSSION THREAD

**Discussion by KamenRiderKuuga** [emqx/emqx#10412] [Q&A]
At our application, there is the concept of gateway devices and actual devices. A gateway device can proxy tens of thousands of devices to publish and subscribe messages. At present, we need to authorize the publish and subscribe operations of the gateway device

Our gateway originally used the wildcard method to subscribe, but now it needs authorization, so it must subscribe to the Topic of each device separately, so what I want to know now is whether there is a difference between the two subscription methods at the implementation of EMQX

**Comment by ieQu1** [comment]
Hello,

Generally, wildcard subscriptions are preferred because they save RAM and it's easier to replicate one routing table record across the cluster than thousands of them. 

I am not sure what is your exact situation is, but you may consider using [MQTT bridge](https://www.emqx.io/docs/en/v5.0/data-integration/data-bridge-mqtt.html#mqtt) feature, or to create an [ACL](https://www.emqx.io/docs/en/v5.0/access-control/authz/file.html#acl-file-format) that specifically allows "gateway" devices to subscribe to wildcard topics.

**Reply by KamenRiderKuuga** [reply]
Sorry, I took it for granted and didn't express it clearly. What I want to achieve is that a so-called "gateway device" only allows subscription to the data of all the devices it manages. 

Our topic looks like this: `data/{deviceId}`, once I want to perform permission control, because the granularity of `data/#` is too large, I cannot subscribe to only some of these devices

**Reply by ieQu1** [reply]
When you're talking about topic structure, does `{deviceId}` refer to the ID of gateway device or device that it manages?
Are `clienid`s and `username`s of the gateway devices known in advance, or are they dynamic?
If gateway devices proxy the traffic, then is it possible to change topic structure to something like `data/{gatewayDeviceClientid}/{managedDeviceClientId}`?
